File: snippets/rust/query_batch_points.rs
async fn main() -> Result<(), qdrant_client::QdrantError> {
    use qdrant_client::qdrant::{Query, QueryBatchPointsBuilder, QueryPointsBuilder};
    use qdrant_client::Qdrant;

    let client = Qdrant::from_url("http://localhost:6334").build()?;

    let request = QueryBatchPointsBuilder::new(
        "{collection_name}",
        vec![
            QueryPointsBuilder::new("")
                .query(Query::new_nearest(vec![0.1; 8]))
                .into(),
            QueryPointsBuilder::new("")
                .query(Query::new_nearest(vec![0.3; 8]))
                .into(),
        ],
    );

    let _response = client.query_batch(request).await?;

    Ok(())
}

File: snippets/python/query_batch_points.py
from qdrant_client import QdrantClient, models

client = QdrantClient(url="http://localhost:6333")

nearest = client.query_batch_points(
    collection_name="{collection_name}",
    requests=[
        models.QueryRequest(
            query=[0.01, 0.45, 0.67],
        ),
        models.QueryRequest(
            query=[0.11, 0.35, 0.6],
        ),
    ]
)